_posts/2021-09-07-dd_00003_logic_gates.md

```diff
@@ -0,0 +1,347 @@
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00003-nand-gate.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+Basically, we just took the circle from the NOT gate and sticked it at the
+output of the AND gate to indicate that the output is negated. We can do the
+same with any other logic gate, such as the OR gate:
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00003-nor-gate.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+This gate is similarly called **NOR** gate. As an excerise, try to write down
+the truth table of a NOR gate!
+
+Let's look at some more examples. Again, first try to figure out the truth table
+of the following circuit by yourself:
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00003-or2-gate.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+Got it? Here is what you should have gotten:
+
+| A   | B   | Z   |
+| --- | --- | --- |
+| 0   | 0   | 0   |
+| 0   | 1   | 1   |
+| 1   | 0   | 1   |
+| 1   | 1   | 1   |
+
+Do you remember this truth table? Right, this is an OR gate! This means, that we
+can construct an OR gate from two NOT gates and one NAND gate. By the way: When
+some inputs of a gate are _inverted_, like in the example above, we don't have
+to draw the full NOT gates. Similar to how we draw a small circle at the output
+of the NAND gate, we can also draw circles at inputs to indicate that these are
+inverted. For instance, here is the simplified version of the example above:
+
+<figure class="img-center" style="width:50%">
+  <img src="/assets/images/dd-00003-or3-gate.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+<!--
+Ealier we said that you only need AND, OR and NOT gates to build any digital circuit. Well, now we can refine that statement: you only need AND and NOT gates! However, this is just relevant in logic theory
+
+And it goes even further. Look at the following circuit:
+
+todo: Picture where same input to NAND (= NOT)
+
+This just a NAND gate where both inputs are connected with each other. This means that both inputs will always have the same value. The truth table of this special NAND gate is just the table of the normal NAND, but without the rows where the inputs are different:
+
+| A | B | Z |
+|---|---|---|
+| 0 | 0 | 1 |
+| 1 | 1 | 0 |
+
+As both inputs are always the same, we can view this NAND gate also as a single input gate; let's just call its input "A". The truth table remains unchanged (just column B disappeared):
+
+| A | Z |
+|---|---|
+| 0 | 1 |
+| 1 | 0 |
+
+Wait, that looks exactly like the TT of the NOT gate!
+-->
+
+Here's another more complex circuit, try to come up (todo: use something else
+than "come up") with its truth table:
```

Comment:
try to find the truth table on your own?

Reply:
1b75418 ?